### PR TITLE
Integrate Social Event Page and Improve Test Coverage of Website Calendar

### PR DESCRIPTION
**Pull request by harshith-venkatesh**
### What is the change?

Created Social Event page, added the design changes for personal events page and addition of test cases

### Is it bug?
No

- Steps to repro
- Expected
- Actual

### \*Dev Tested?

- [X] Yes
- [ ] No

#### Platforms

- [X] Web

#### Browsers

- [X] Chrome
- [X] Safari
- [X] Firefox

### Before / After Change Screenshots
Screenshot 2023-06-19 at 5 17 05 AM

### CODE REVIEW

**Comment by isVivek99** [comment]
i cannot see any changed test file, also some breaking tests in eventCard @harshith-venkatesh , did u forget to push anything?

File: app/components/common/socialEventCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+export default function SocialEventCard({ title, attendees, start, end }: CalEvent) {
+  const startDate = dayjs(start);
+  const date = startDate.format('D');
+  const month = startDate.format('MMM');
+  const startTime = startDate.format('h:mm A');
+  const endTime = dayjs(end).format('h:mm A');
+  const numberOfAttendees = attendees ? attendees.length : 0;
+  return (
+    <>
+      <div className="border-2 w-64 h-72 bg-gray-200 rounded-3xl m-auto p-4 flex flex-col justify-between">
```
Comment (iRohitGaur) [inline app/components/common/socialEventCard/index.tsx:13]: `cursor: pointer;` required

File: app/routes/events.tsx [hunk 2]
```diff
@@ -23,22 +26,26 @@ const Events = () => {
...
-            <h2>{event.title}</h2>
-            <p>You and John</p>
+      <div className="">
+      <div className="flex flex-col-reverse md:flex-row ">
+        <Navbar />
+        <div className="flex justify-center flex-grow p-5 items-start">
+          <div className="w-full grid grid-cols-1 gap-2">
+            <div className="w-full text-center text-grey-med">Events</div>
+            <div className="w-full grid grid-cols-1 gap-2">
+              {dummyEvent.map(eventData => <div onClick={() => toggleDrawer(eventData)}><EventCard {...eventData} /></div>)}
```
Comment (iRohitGaur) [inline app/routes/events.tsx:43]: `cursor: pointer;` required

**Comment by iRohitGaur** [comment]
minor issue with the navbar as discussed on the call

File: public/mockServiceWorker.js [hunk 3]
```diff
@@ -250,53 +251,53 @@ async function getResponse(event, client, requestId) {
...
-    setTimeout(resolve, timeMs);
-  });
+    setTimeout(resolve, timeMs)
+  })
 }
 
 async function respondWithMock(response) {
-  await sleep(response.delay);
-  return new Response(response.body, response);
+  await sleep(response.delay)
```
Comment (isVivek99) [inline public/mockServiceWorker.js:301]: how did semicolons get added?

File: app/routes/events.tsx [hunk 4]
```diff
@@ -1,6 +1,8 @@
 import dayjs from 'dayjs';
```
Comment (isVivek99) [inline app/routes/events.tsx:1]: are we keeping the events static as of now?
for both `events` & `socialEvents` routes?

**Review by isVivek99 — CHANGES_REQUESTED** [review]

File: app/components/common/drawer/index.tsx [hunk 5]
```diff
@@ -34,7 +34,7 @@ const Drawer: React.FC<DrawerProps> = ({ event, isDrawerVisible, toggleDrawer })
           <h1 className=" font-semibold text-3xl leading-10 capitalize">{event.title}</h1>
           <EventVisibility visibility={event.visibility ?? ''} />
           <p className="py-2 text-base font-normal leading-5 text-left">
-            {moment(event.start).format('MMMM DD, YYYY h A')} - {moment(event.end).format('h A')}
```
Comment (isVivek99) [inline app/components/common/drawer/index.tsx:37]: can we remove moment and add dayjs instead

File: app/components/common/drawer/index.test.tsx [hunk 6]
```diff
@@ -55,7 +55,7 @@ describe('Drawer', () => {
     expect(toggleDrawerMock).toHaveBeenCalledTimes(1);
 
     const eventTime = screen.getByText(
-      `${moment(event.start).format('MMMM DD, YYYY h A')} - ${moment(event.end).format('h A')}`,
+      `${moment(event.start).format('MMMM DD, YYYY h A')} - ${moment(event.end).format('MMMM DD, YYYY h A')}`,
```
Comment (isVivek99) [inline app/components/common/drawer/index.test.tsx:58]: can we move from moment to dayjs

**Comment by isVivek99** [comment]
will we bring tests back for components which were deleted?

File: app/components/common/socialEventCard/index.tsx (hunk 1 above)
Comment (harshith-venkatesh) [reply app/components/common/socialEventCard/index.tsx]: resolved @iRohitGaur

File: app/routes/events.tsx (hunk 2 above)
Comment (harshith-venkatesh) [reply app/routes/events.tsx:43]: resolved @iRohitGaur

File: public/mockServiceWorker.js (hunk 3 above)
Comment (harshith-venkatesh) [reply public/mockServiceWorker.js:301]: I am not sure, this has been fixed

File: app/routes/events.tsx (hunk 4 above)
Comment (harshith-venkatesh) [reply app/routes/events.tsx:1]: Currently yes, when we have clear private, public and social events are created we are replace accordingly

File: app/components/common/drawer/index.tsx (hunk 5 above)
Comment (harshith-venkatesh) [reply app/components/common/drawer/index.tsx:37]: updated

File: app/components/common/drawer/index.test.tsx (hunk 6 above)
Comment (harshith-venkatesh) [reply app/components/common/drawer/index.test.tsx]: addressed

File: app/components/common/navbar/__test__/Navbar.test.tsx
Comment (harshith-venkatesh) [inline app/components/common/navbar/__test__/Navbar.test.tsx:1]: Creating a separate ticket to address the issue related to Navbar test files, Slider Test File

Comment (harshith-venkatesh) [reply app/components/common/navbar/__test__/Navbar.test.tsx:1]: https://github.com/Real-Dev-Squad/website-calendar/issues/164

**Comment by harshith-venkatesh** [comment]
https://github.com/Real-Dev-Squad/website-calendar/issues/164 to address the failed test cases that are currently being removed

File: app/routes/events.tsx (hunk 4 above)
Comment (isVivek99) [reply app/routes/events.tsx:1]: makes sense

**Review by isVivek99 — APPROVED** [review]

File: app/components/common/navbar/__test__/Navbar.test.tsx
Comment (isVivek99) [reply app/components/common/navbar/__test__/Navbar.test.tsx:1]: ok will check that

**Review by iRohitGaur — APPROVED** [review]
Just to note so we don't forget: navbar highlight issue is still there. Route is not highlighted in nav unless you click it again. Can be addressed later as well.

Rest all good.

**Comment by harshith-venkatesh** [comment]
Thanks for mentioning it @iRohitGaur , I will create a ticket to track that

**Pull request merged**